aws_crt_common_ws/aws-crt-c-flags/src/lib.rs

```diff
@@ -0,0 +1,469 @@
+        );
+        let target_include_path = Path::new(out_dir.as_str());
+        fs::create_dir_all(target_include_path).expect("Creation of directory failed!");
+
+        let mut terrible_api_hack = vec![];
+        terrible_api_hack.push(dir);
+
+        let mut copy_options = CopyOptions::new();
+        copy_options.overwrite = true;
+
+        fs_extra::copy_items(
+            terrible_api_hack.as_ref(),
+            target_include_path,
+            &copy_options,
+        )
+        .expect("Copy failed, check the directory exists");
+        self.include_dirs.push(target_include_path.to_path_buf());
+
+        self
+    }
+
+    /// Writes generated content to your build directory, for use with things like autoconf output.
+    ///
+    /// # Arguments
+    ///
+    /// * `generated_content` - The file contents you want to write.
+    /// * `to` - Location relative to your build tree you'd like to write to.
+    ///
+    ///# Examples
+    ///```should_panic
+    /// use aws_crt_c_flags::{CRTModuleBuildInfo};
+    /// use std::path::Path;
+    /// let mut build_info = CRTModuleBuildInfo::new("aws_crt_common_sys");
+    /// let content = "test content".to_string();
+    /// build_info.write_generated_file_to_output_path(&content, Path::new("include/aws/common/config.h"));
+    ///```
+    pub fn write_generated_file_to_output_path(
+        &mut self,
+        generated_content: &String,
+        to: &Path,
+    ) -> &mut CRTModuleBuildInfo {
+        let target_location: String = format!(
+            "{}/{}",
+            env::var_os("OUT_DIR").unwrap().to_str().unwrap(),
+            to.display()
+        );
+        let target_path = Path::new(target_location.as_str());
+        fs::create_dir_all(target_path.parent().unwrap())
+            .expect("Creation of output directory failed.");
+        fs::write(target_location, generated_content).expect("Writing generated file failed!");
+
+        self
+    }
+
+    /// Returns the underlying toolchain that will be used for the build.
+    pub fn get_toolchain(&self) -> &Build {
+        &self.build_toolchain
+    }
+
+    /// Adds the file at path to the build tree
+    pub fn add_file_to_build(&mut self, path: &Path) -> &mut CRTModuleBuildInfo {
+        self.build_toolchain.file(path);
+        self
+    }
+
+    /// Attempts to compile, `to_compile` and returns a result on whether or not it succeeded.
+    /// This is useful for testing compiler capabilities before including a file or flag in your build.
+    ///
+    /// # Arguments
+    ///
+    /// * `to_compile` - C code to attempt compilation of.
+    ///
+    /// # Examples
+    /// ```should_panic
+    /// use aws_crt_c_flags::{CRTModuleBuildInfo};
+    /// let mut build_info = CRTModuleBuildInfo::new("aws_crt_common_sys");
+    /// build_info.try_compile("int main() { return 0; }").expect("This should have compiled");
+    /// ```
+    pub fn try_compile(&self, to_compile: &str) -> core::result::Result<(), cc::Error> {
```

Comment:
I found the `core` usage here confusing at first. Usually `core` is referenced when `std` is not being used. `std::result::Result` is a re-export of `core::result::Result` so they are the same types.

Reply:
that was a clion insert thing. I'll move it over. good catch!